### PR TITLE
support Klusterlet.Spec.Namespace in hosted mode

### PR DESCRIPTION
**Pull request by elgnay**
Signed-off-by: Yang Le <yangle@redhat.com>

### CODE REVIEW

File: pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go [hunk 1]
```diff
@@ -143,8 +143,15 @@ func NewKlusterletController(
 
 // klusterletConfig is used to render the template of hub manifests
 type klusterletConfig struct {
-	KlusterletName            string
-	KlusterletNamespace       string
+	KlusterletName string
+	// KlusterletNamespace is the klusterlet namespace on the managed cluster.
```
Comment (zhujian7) [inline pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go]: Can we add more comments on this to explain something like:
- in default mode, it is identical to the agent namespace.
- in hosted mode, we store some configuration in this namespace that will be used by the agents on the remote management cluster.

Comment (elgnay) [reply pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go]: Doc updated for those two fields.

**Review by qiujian16 — COMMENTED** [review]
/approve

**Pull request merged**